Runtime issue: cannot use KameletBinding error handler

It appears the feature is broken when using either `1.5` or `latest`:
```
[1] 2021-07-23 09:28:42,268 ERROR [org.apa.cam.qua.mai.CamelMainRuntime] (main) Failed to start application: org.apache.camel.FailedToCreateRouteException: Failed to create route route1 at: >>> To[log:bar] <<< in route: Route(route1)[From[timer:foo] -> [To[log:bar]]] because of No bean could be found in the registry for:  defaultErrorHandler of type: org.apache.camel.ErrorHandlerFactory
[1] 	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:240)
[1] 	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:74)
[1] 	at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
[1] 	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:826)
[1] 	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:716)
[1] 	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2756)
[1] 	at org.apache.camel.quarkus.core.FastCamelContext.doInit(FastCamelContext.java:158)
[1] 	at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
[1] 	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2475)
[1] 	at org.apache.camel.support.service.BaseService.start(BaseService.java:111)
[1] 	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2494)
[1] 	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:245)
[1] 	at org.apache.camel.quarkus.main.CamelMain.doStart(CamelMain.java:94)
[1] 	at org.apache.camel.support.service.BaseService.start(BaseService.java:119)
[1] 	at org.apache.camel.quarkus.main.CamelMain.startEngine(CamelMain.java:139)
[1] 	at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:49)
[1] 	at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
[1] 	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy_0(CamelBootstrapProcessor$boot-173480958.zig:101)
[1] 	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy(CamelBootstrapProcessor$boot-173480958.zig:40)
[1] 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:537)
[1] 	at io.quarkus.runtime.Application.start(Application.java:101)
[1] 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:101)
[1] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
[1] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
[1] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
[1] 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
[1] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[1] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[1] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[1] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[1] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:48)
[1] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:25)
[1] Caused by: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for:  defaultErrorHandler of type: org.apache.camel.ErrorHandlerFactory
[1] 	at org.apache.camel.support.CamelContextHelper.mandatoryLookup(CamelContextHelper.java:241)
[1] 	at org.apache.camel.model.errorhandler.ErrorHandlerHelper.lookupErrorHandlerFactory(ErrorHandlerHelper.java:82)
[1] 	at org.apache.camel.reifier.errorhandler.ErrorHandlerRefReifier.lookupErrorHandler(ErrorHandlerRefReifier.java:41)
[1] 	at org.apache.camel.reifier.errorhandler.ErrorHandlerRefReifier.createErrorHandler(ErrorHandlerRefReifier.java:35)
[1] 	at org.apache.camel.impl.DefaultModelReifierFactory.createErrorHandler(DefaultModelReifierFactory.java:65)
[1] 	at org.apache.camel.reifier.ProcessorReifier.wrapInErrorHandler(ProcessorReifier.java:745)
[1] 	at org.apache.camel.reifier.ProcessorReifier.wrapChannelInErrorHandler(ProcessorReifier.java:726)
[1] 	at org.apache.camel.reifier.ProcessorReifier.wrapChannel(ProcessorReifier.java:705)
[1] 	at org.apache.camel.reifier.ProcessorReifier.wrapChannel(ProcessorReifier.java:611)
[1] 	at org.apache.camel.reifier.ProcessorReifier.wrapProcessor(ProcessorReifier.java:607)
[1] 	at org.apache.camel.reifier.ProcessorReifier.makeProcessor(ProcessorReifier.java:854)
[1] 	at org.apache.camel.reifier.ProcessorReifier.addRoutes(ProcessorReifier.java:579)
[1] 	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:236)
[1] 	... 31 more
```
I think it's some change of type within the `camel-k-runtime` latest upgrade. Not yet sure though, it will need more investigation.

--- a/pkg/apis/camel/v1/integration_types_support.go
+++ b/pkg/apis/camel/v1/integration_types_support.go
@@ -148,11 +148,18 @@ func (in *IntegrationSpec) GetConfigurationProperty(property string) string {
 		if confSpec.Type == "property" && strings.HasPrefix(confSpec.Value, property) {
 			splitConf := strings.Split(confSpec.Value, "=")
 			if len(splitConf) > 1 {
-				return splitConf[1]
+				return trimFirstLeadingSpace(splitConf[1])
 			}
 		}
 	}
 	return ""
+}
+
+func trimFirstLeadingSpace(val string) string {
+	if strings.HasPrefix(val, " ") {
+		return val[1:]
+	}
+	return val
 }
 
 // AddOrReplaceGeneratedResources --

--- a/pkg/apis/camel/v1/integration_types_support_test.go
+++ b/pkg/apis/camel/v1/integration_types_support_test.go
@@ -73,3 +73,28 @@ func TestAddDependency(t *testing.T) {
 	integration.AddDependency("file:dep")
 	assert.Equal(t, integration.Dependencies, []string{"file:dep"})
 }
+
+func TestGetConfigurationProperty(t *testing.T) {
+	integration := IntegrationSpec{}
+	integration.AddConfiguration("property", "key1=value1")
+	integration.AddConfiguration("property", "key2 = value2")
+	integration.AddConfiguration("property", "key3 = value with trailing space ")
+	integration.AddConfiguration("property", "key4 =  value with leading space")
+	integration.AddConfiguration("property", "key5 = ")
+	integration.AddConfiguration("property", "key6=")
+
+	missing := integration.GetConfigurationProperty("missing")
+	assert.Equal(t, "", missing)
+	v1 := integration.GetConfigurationProperty("key")
+	assert.Equal(t, "value1", v1)
+	v2 := integration.GetConfigurationProperty("key2")
+	assert.Equal(t, "value2", v2)
+	v3 := integration.GetConfigurationProperty("key3")
+	assert.Equal(t, "value with trailing space ", v3)
+	v4 := integration.GetConfigurationProperty("key4")
+	assert.Equal(t, " value with leading space", v4)
+	v5 := integration.GetConfigurationProperty("key5")
+	assert.Equal(t, "", v5)
+	v6 := integration.GetConfigurationProperty("key6")
+	assert.Equal(t, "", v6)
+}

